Makefile

```diff
@@ -1,91 +1,167 @@
+# See https://en.wikipedia.org/wiki/ANSI_escape_code#SGR_(Select_Graphic_Rendition)_parameters
+b_black            := $(shell printf "\33[40m")
+f_white            := $(shell printf "\33[97m")
+f_gray             := $(shell printf "\33[37m")
+f_dark_gray        := $(shell printf "\33[90m")
+f_bright_magenta   := $(shell printf "\33[95m")
+b_bright_magenta   := $(shell printf "\33[105m")
+ansi_reset         := $(shell printf "\33[0m")
+ansi_car           := $(b_black)$(f_white)car$(ansi_reset)
+ansi_format_dark   := $(f_gray)$(f_bright_magenta)%-10s$(ansi_reset) $(f_dark_gray)%s$(ansi_reset)\n
+ansi_format_bright := $(f_white)$(f_bright_magenta)%-10s$(ansi_reset) $(f_white)$(b_bright_magenta)%s$(ansi_reset)\n
+
+# This formats help statements in ANSI colors. To hide a target from help, don't comment it with a trailing '##'.
+help: ## Describe how to use each target
+	@printf "$(ansi_car)$(f_white)\n"
+	@awk 'BEGIN {FS = ":.*?## "} /^[0-9a-zA-Z_-]+:.*?## / {sub("\\\\n",sprintf("\n%22c"," "), $$2);printf "$(ansi_format_dark)", $$1, $$2}' $(MAKEFILE_LIST)
+
+build: $(current_binary) ## Build the car binary
+
+test_packages := . ./internal/...
+test: ## Run all unit tests
+	@printf "$(ansi_format_dark)" test "running unit tests"
+	@$(go) test $(test_packages)
+	@printf "$(ansi_format_bright)" test "ok"
+
+coverage: ## Generate test coverage
+	@printf "$(ansi_format_dark)" coverage "running unit tests with coverage"
+	@$(go) test -coverprofile=coverage.txt -covermode=atomic --coverpkg $(test_packages: =,) $(test_packages)
+	@$(go) tool cover -func coverage.txt
+	@printf "$(ansi_format_bright)" coverage "ok"
+
+platforms := darwin_amd64 darwin_arm64 linux_amd64 linux_arm64
+
+# Make 3.81 doesn't support '**' globbing: Set explicitly instead of recursion.
+all_patterns := *.go */*.go */*/*.go */*/*/*.go */*/*/*.go */*/*/*/*.go
+all_sources  := $(wildcard $(all_patterns))
+main_sources := $(wildcard $(subst *,*[!_test],$(all_patterns)))
+
+build/car_%/car: $(main_sources)
+	$(call go-build, $@, $<)
+
+dist/car_$(VERSION)_%.tar.gz: build/car_%/car
+	@printf "$(ansi_format_dark)" tar.gz "tarring $@"
+	@mkdir -p $(@D)
+	@tar --strip-components 2 -cpzf $@ $<
+	@printf "$(ansi_format_bright)" tar.gz "ok"
+
+archives  := $(platforms:%=dist/car_$(VERSION)_%.tar.gz)
+checksums := dist/car_$(VERSION)_checksums.txt
+
+# Darwin doesn't have sha256sum. See https://github.com/actions/virtual-environments/issues/90
+sha256sum := $(if $(findstring darwin,$(goos)),shasum -a 256,sha256sum)
+$(checksums): $(archives) $(packages)
+	@printf "$(ansi_format_dark)" sha256sum "generating $@"
+	@$(sha256sum) $^ > $@
+	@printf "$(ansi_format_bright)" sha256sum "ok"
+
+# dist generates the assets that attach to a release
+# Ex. https://github.com/tetratelabs/car/releases/tag/v$(VERSION)
+dist: $(archives) $(packages) $(checksums) ## Generate release assets
+
+clean: ## Ensure a clean build
+	@printf "$(ansi_format_dark)" clean "deleting temporary files"
```

Comment:
nit: `@printf "$(ansi_format_xxxx)"` seems repetitive. Can't be in a func like `xxxx`??